cheatsheets/Credential_Stuffing_Prevention_Cheat_Sheet.md

```diff
@@ -10,13 +10,21 @@ It should be noted that defense mechanisms are intended to be used in a layered
 
 In many cases, brute force protections will overlap with credential stuffing defenses.
 
+Keep in mind that application can have different security levels for different users/roles or actions. For example for casual customers multi-factor authentication may by optional but for admins or some other special roles it can be required.
+
 ## Defense Option 1: Multi-Factor Authentication
 
 True multi-factor authentication is the best defense against Credential Stuffing attacks, but can have significant deployment and usability impacts, and so is frequently not a feasible option. If this defense is not feasible for your application, consider adopting as many of these other defenses as you can.
 
-## Defense Option 2: Multi-Step Login Process
+In order to balance security and usability multi-factor authentication can be combined with other techniques to ask for 2nd factor only in special situations. For example it can be combined with [device fingerprinting](cheatsheets/Credential_Stuffing_Prevention_Cheat_Sheet.md#defense-option-4-device-fingerprinting), in that scenario application can ask for 2nd factor only when application is accessed by unknown, new browser. Similary, cookies can be also used for remembering known browsers or application can ask about another factor based on login success ratio, IP address (like users from different network than company one like remote workers) etc.
```

Comment:
`and usability,` add the comma. `For example,`. `in that scenario, the application can ask for a 2nd factor only when the application is accessed by an unknown, new browser.`. `cookies can also be used`. `or the application can ask about another factor based on the login`.